apps/teams-test-app/src/components/privateApis/MeetingRoomAPIs.tsx

```diff
@@ -1,104 +1,80 @@
-      .catch(error => setGetPairedMeetingRoomInfoRes('Error code: ' + error));
-  };
+const SendCommandToPairedMeetingRoom = (): React.ReactElement =>
+  ApiWithTextInput<string>({
+    name: 'sendCommandToPairedMeetingRoom',
+    title: 'Send Command To PairedMeeting Room',
+    onClick: {
+      validateInput: input => {
+        if (!input) {
+          throw new Error('input is required.');
+        }
+      },
+      submit: async input => {
+        await meetingRoom.sendCommandToPairedMeetingRoom(input);
+        return 'sendCommandToPairedMeetingRoom have been called';
+      },
+    },
+  });
 
-  const sendCommandToPairedMeetingRoom = (commandName: string): void => {
-    setSendCommandToPairedMeetingRoomRes('meetingRoom.sendCommandToPairedMeetingRoom' + noHostSdkMsg);
-    meetingRoom
-      .sendCommandToPairedMeetingRoom(commandName)
-      .then(() => setSendCommandToPairedMeetingRoomRes('sendCommandToPairedMeetingRoom have been called'))
-      .catch(error => setSendCommandToPairedMeetingRoomRes('Error code: ' + error));
-  };
-  const registerMeetingRoomCapabilitiesUpdateHandler = (): void => {
-    setRegisterMeetingRoomCapabilitiesUpdateHandlerRes(
-      generateRegistrationMsg('it is invoked when the live stream state changes'),
-    );
-    const handler = (meetingRoomCapability: meetingRoom.MeetingRoomCapability): void => {
-      setRegisterMeetingRoomCapabilitiesUpdateHandlerRes(
-        `Capabilities of meeting room update ${JSON.stringify(meetingRoomCapability)}`,
-      );
-    };
-    meetingRoom.registerMeetingRoomCapabilitiesUpdateHandler(handler);
-  };
-  const registerMeetingRoomStatesUpdateHandler = (): void => {
-    setRegisterMeetingRoomStatesUpdateHandlerRes(
-      generateRegistrationMsg('it is invoked when the live stream state changes'),
-    );
-    const handler = (meetingRoomState: meetingRoom.MeetingRoomState): void => {
-      setRegisterMeetingRoomStatesUpdateHandlerRes(`States of meeting room update ${JSON.stringify(meetingRoomState)}`);
-    };
-    meetingRoom.registerMeetingRoomStatesUpdateHandler(handler);
-  };
+const RegisterMeetingRoomCapabilitiesUpdateHandler = (): React.ReactElement =>
+  ApiWithoutInput({
+    name: 'registerMeetingRoomCapabilitiesUpdateHandler',
+    title: 'Register MeetingRoom Capabilities Update Handler',
+    onClick: async setResult => {
+      const handler = (meetingRoomCapability: meetingRoom.MeetingRoomCapability): void => {
+        setResult(`Capabilities of meeting room update ${JSON.stringify(meetingRoomCapability)}`);
+      };
+      meetingRoom.registerMeetingRoomCapabilitiesUpdateHandler(handler);
 
-  const checkMeetingRoomCapability = (): void => {
-    if (meetingRoom.isSupported()) {
-      setCapabilityCheckRes('MeetingRoom is supported');
-    } else {
-      setCapabilityCheckRes('MeetingRoom is not supported');
-    }
-  };
+      return generateRegistrationMsg('it is invoked when the live stream state changes');
```

Comment:
Another example that I know is just carried over from previous code, but I think this was meant to be something like 'the meeting room capabilities update'. (the generateRegistration message returns this, so the "it is invoked" is also not necessary: `Registration attempt has been initiated. If successful, this message will change when ${changeCause}.` But I don't think it's a big deal if you leave the double "when"s for this message or the others)